software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/ContainerTaskFactory.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.brooklyn.util.core.task.system.ProcessTaskStub;
+import org.apache.brooklyn.util.core.task.system.ProcessTaskWrapper;
+import org.apache.brooklyn.util.core.task.system.internal.SystemProcessTaskFactory;
+import org.apache.brooklyn.util.text.Strings;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.brooklyn.tasks.kubectl.ContainerCommons.*;
+
+public class ContainerTaskFactory<T extends ContainerTaskFactory<T,RET>,RET>  implements TaskFactory<Task<RET>> {
+
+    protected String summary;
+    protected String tag = "";
+    protected final ConfigBag config = ConfigBag.newInstance();
+
+    @Override
+    public Task<RET> newTask() {
+        ConfigBag configBag = this.config;
+
+        List<String> commandsCfg =  EntityInitializers.resolve(configBag, COMMANDS);
+        List<String> argumentsCfg =  EntityInitializers.resolve(configBag, ARGUMENTS);
+        String containerImage = EntityInitializers.resolve(configBag, CONTAINER_IMAGE);
+        String containerNameFromCfg = EntityInitializers.resolve(configBag, CONTAINER_NAME);
+        Boolean devMode = EntityInitializers.resolve(configBag, DEV_MODE);
+
+        if(Strings.isBlank(containerImage)) {
+            throw new IllegalStateException("You must specify containerImage when using " + this.getClass().getSimpleName());
+        }
+
+
+        final String containerName = (Strings.isBlank(containerNameFromCfg)
+                ? ( (Strings.isNonBlank(this.tag) ? this.tag + "-" : "").concat(containerImage).concat("-").concat(Strings.makeRandomId(10)))
+                : containerNameFromCfg).replace(" ", "-").replace("/", "-").toLowerCase();
+
+        final String jobYamlLocation =  new JobBuilder()
+                .withImage(containerImage)
+                .withName(containerName)
+                .withArgs(argumentsCfg)
+                .withEnv(EntityInitializers.resolve(configBag, BrooklynConfigKeys.SHELL_ENVIRONMENT))
+                .withCommands(Lists.newArrayList(commandsCfg))
+                .build();
+
+
+        Task<String> runCommandsTask = buildKubeTask(configBag, "Submit job", String.format(JOBS_CREATE_CMD,jobYamlLocation)).asTask();
+        Task<String> waitTask =  buildKubeTask(configBag, "Wait For Completion", String.format(JOBS_FEED_CMD,containerName)).asTask();
+        if(!devMode) {
+            BrooklynTaskTags.addTagDynamically(waitTask, BrooklynTaskTags.INESSENTIAL_TASK);
+        }
+
+        TaskBuilder<RET> taskBuilder = Tasks.<RET>builder()
+                .displayName(this.summary)
+                .add(buildKubeTask(configBag, "Set Up and Run",
+                        String.format(NAMESPACE_CREATE_CMD,containerName),
+                        String.format(NAMESPACE_SET_CMD,containerName)))
+                .add(runCommandsTask)
+                .add(waitTask)
+                .add(buildKubeTask(configBag, "Retrieve Output", String.format(JOBS_LOGS_CMD,containerName)));
+
+        if(!devMode) {
```

Comment:
My feeling about `devMode` is it could be a risk allow not to remove a sensor container based on a blueprint parameter. It will create a new container periodically and if is not removed will consume resources and pull down the kubernetes cluster eventually.

Keeping that property in the `booklyn.cfg` file let the Brooklyn admin to allow or not that behaviour at the server level instead the blueprints and I think is safer -> dev server allow dev blueprints in a dev kubernetes cluster, but the production server doesn't

Reply:
I would rename `devMode` as `keepContainersForDebugging` with an additional comment in description that if true, the operator must take responsibility for cleaning up containers.  It should be easy enough, e.g. a cron job which deletes finished containers matching the brooklyn container naming convention, no?

So I don't think we yet need a global property to forbid this, though that could be an enhancement in future.

Reply:
@ahgittin  renamed to `keepContainerForDebugging`